validator/sawtooth_validator/server/cli.py

```diff
@@ -150,31 +144,78 @@ def main(path_config, validator_config, verbosity):
             ])
         )
 
+
+def check_directories(path_config):
+    init_errors = False
+
     if not check_directory(path=path_config.data_dir,
                            human_readable_name='Data'):
         init_errors = True
     if not check_directory(path=path_config.log_dir,
                            human_readable_name='Log'):
         init_errors = True
 
-    endpoint = validator_config.endpoint
-    if endpoint is None:
-        # Need to use join here to get the string "0.0.0.0". Otherwise,
-        # bandit thinks we are binding to all interfaces and returns a
-        # Medium security risk.
-        interfaces = ["*", ".".join(["0", "0", "0", "0"])]
-        interfaces += netifaces.interfaces()
-        endpoint = validator_config.bind_network
-        for interface in interfaces:
-            if interface in validator_config.bind_network:
-                LOGGER.error("Endpoint must be set when using %s", interface)
-                init_errors = True
-                break
+    return init_errors
+
 
+def exit_if_errors(init_errors):
     if init_errors:
         LOGGER.error("Initialization errors occurred (see previous log "
                      "ERROR messages), shutting down.")
         sys.exit(1)
+
+
+def check_interfaces(endpoint):
+    # Need to use join here to get the string "0.0.0.0". Otherwise,
+    # bandit thinks we are binding to all interfaces and returns a
+    # Medium security risk.
+    interfaces = ["*", ".".join(["0", "0", "0", "0"])]
+    interfaces += netifaces.interfaces()
+
+    for interface in interfaces:
+        if interface in endpoint:
+            LOGGER.error("Endpoint must be set when using %s", interface)
+            return True
+
+    return False
```

Comment:
Careful, this code is being modified here: https://github.com/hyperledger/sawtooth-core/pull/1592